tools/phenomenal/ms/save_chromatogram.xml

```diff
@@ -1,14 +1,20 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <!--Proposed Tool Section: [Statistics]-->
-<tool id="save_chromatogram" name="save_chromatogram" version="1.1">
+<tool id="save_chromatogram" name="save_chromatogram" version="1.2">
 <!--
   <requirements>
-    <container type="docker">phnmnl/xcms</container>
+    <container type="docker">container-registry.phenomenal-h2020.eu/phnmnl/xcms</container>
   </requirements>
 -->
-  <description>Saves the chromatogram of a single mzML file.</description>
+  <description>Saves the chromatogram in CSV format of a single mzML file.</description>
   <command><![CDATA[
-    save_chromatogram.r $infile $outfile 2>&1
+if [[ "\$(head -n 2 $infile | tail -n 1 | grep mzML)" != "" ]];
+then filename="\$(head -n 40 $infile | grep "sample id" | sed -e "s/\"\>.*//" | sed -e "s/.*name\=\"//")";
+    xcmsfile="/tmp/\$filename.mzML";
+    cp $infile \$xcmsfile;
+    save_chromatogram.r \$xcmsfile $outfile 2>&1;
+else echo "Error! Input file is not mzML.";
```

Comment:
If wanting to show an error, you probably need to exit with a non zero code I presume. Otherwise, I don't think Galaxy will show this as a failed job in red (which is what you probably want if the user connected an non mzML file).